dependencies.yaml

```diff
@@ -306,7 +310,7 @@ dependencies:
           - numpy>=1.23,<3.0a0
           - zarr
           # See https://github.com/zarr-developers/numcodecs/pull/475
-          - numcodecs <0.12.0
+          - numcodecs !=0.12.0
```

Comment:
`kvikio` currently pins `numcodecs<12.0`

https://github.com/rapidsai/kvikio/blob/0e9ebcbc74b83302c8b73a8ac0b1ef6555919263/dependencies.yaml#L308-309

No version in that range published Python 3.12 builds on `conda-forge`.

```text
- Encountered problems while solving:
-   - package numcodecs-0.10.0-py310hd8f1fbe_0 requires python_abi 3.10.* *_cp310, but none of the providers can be installed
```

([build link](https://github.com/rapidsai/kvikio/actions/runs/10707730321/job/29688643926?pr=458#step:7:569))

Looks like that pin was added in #300. The corresponding fix (https://github.com/zarr-developers/numcodecs/pull/475) has been merged upstream and made it into `numcodecs==0.12.1`, so hopefully this pin can be relaxed now.